src/Message/FetchTransactionRequest.php

```diff
@@ -11,6 +11,8 @@
      */
     public function getData()
     {
+        $this->validate('transactionReference');
```

Comment:
That is indeed the desired behaviour: `This method is called internally by gateways to avoid wasting time with an API call when the request is clearly invalid`